forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,258 @@
+
+    def _parse_pack(
+        self, line_info: Tuple[int, List[str]]
+    ) -> Iterator[DataPack]:
+        line_id, line = line_info
+        # content label title
+        pack = DataPack()
+
+        if len(self.configs.forte_data_fields) != len(line):
+            raise ProcessorConfigError(
+                "Forte data fields provided in config "
+                "doesn't have one-to-one correspondence with the actual data "
+                "fields from dataset.\n"
+                f"Data fields length: {len(self.configs.forte_data_fields)} \n"
+                f"Line length: {len(line)}"
+            )
+        df_dict = dict(zip(self.configs.forte_data_fields, line))
+
+        # it determines the order of concatenation
+        text_fields = self.configs.text_fields
+
+        # get text and ontology indices
+        text, input_ontology_indices = generate_text_n_input_ontology_indices(
+            text_fields, df_dict
+        )
+        pack.set_text(text)
+        # self.set_text(pack, text)
+        if df_dict["label"].isdigit() != self.configs.digit_label:
+            dataset_digit_label = df_dict["label"].isdigit()
+            raise ProcessorConfigError(
+                "Label format from dataset "
+                "is not consistent with the label format from configs. \n"
+                f"dataset digit label: {dataset_digit_label}"
+                f"config digit label: {self.configs.digit_label}",
+            )
+
+        if self.configs.digit_label:
+            if self.configs.one_based_index_label:
+                # adjust label encoding if dataset digit label is one-based
+                class_id = int(df_dict["label"]) - 1
+            else:
+                class_id = int(df_dict["label"])
+        else:
+            # decode string label from dataset into digit label
+            class_id = self._class2index[df_dict["label"]]
+        # initialize all ontologies instances as data pack module
+        # to add corresponding variables and functions
+        for (
+            ontology_field,
+            (start_idx, end_idx),
+        ) in input_ontology_indices.items():
+            path_str, module_str = ontology_field.rsplit(".", 1)
+            mod = importlib.import_module(path_str)  # sentence ontology module
+            entry_class = getattr(mod, module_str)
+            entry_class(pack, start_idx, end_idx)
+        # for now, we use document to store concatenated text and set the class here
+        doc = Document(pack, 0, input_ontology_indices[text_fields[-1]][1])
+        doc.document_class = [self.configs.index2class[class_id]]
+
+        pack.pack_name = line_id
+        yield pack
+
+    @classmethod
+    def default_configs(cls):
+        r"""This defines a basic configuration structure for classification dataset reader.
+
+        Here:
+            - forte_data_fields: these fields provides one-to-one
+            correspondence between given original dataset column names and
```

Comment:
the indent here should be wrong, could you try to build the sphinx doc and double check the format

Reply:
Is it expecting some errors if it's wrong? I changed the format to double spaces instead of column starting from `Here` column. And use `sphinx-build -W -b html -d _build/doctrees . _build/html` before and after changing the format doesn't make any difference.

Reply:
no unfortunately there is no error (an error would make this easier). basically, we would want the documentation to look good. Based on some of my experience, if the word `correspond` is not indented correctly (now it is indented at the same level of the `-` symbol), it will break the list structure. Can you maybe post a screenshot of the documentation you generated here so that I can take a look?

Reply:
Here is an example of the documentation that's rendered terribly: https://asyml-forte.readthedocs.io/en/latest/code/data.html#forte.data.data_pack.DataPack.get_original_span, you can see from the [source](https://asyml-forte.readthedocs.io/en/latest/_modules/forte/data/data_pack.html#DataPack.get_original_span), the docstring looks fine, but the indent is wrong, which generated messy documentation.

Reply:
I found other reader docs generated but not the `classification_reader.py`. Is there any common reason for it?

Reply:
oh, i see, I always forget about this. We need to edit one file to include this: https://github.com/asyml/forte/blob/master/docs/code/data.rst

Reply:
I render it `classification_reader.html` through online html viewer. It looks like this. I am not sure how to render it locally like https://asyml-forte.readthedocs.io/en/latest/code/data.html 

![image](https://user-images.githubusercontent.com/28021889/151648102-fe3225c1-0678-4375-812f-86937ebd0cef.png)

Reply:
So you have the HTML files ready? What do you mean by rendering locally? You can simply view HTML pages the with browser.

Reply:
This is the new rendering result that I used the method I mentioned in the meeting.
![image](https://user-images.githubusercontent.com/28021889/151892098-0e4aa83b-da2d-4c2e-bfc3-52b204eaef22.png)